test/Array/main.cpp

```diff
@@ -133,26 +138,37 @@ static void test_non_pod() {
     }
     for (idx = 0; idx < initial_capacity; idx ++) {
         const Test& t = array[idx];
-        MBED_HOSTTEST_ASSERT(t == Test(idx, 'b'));
+        TEST_ASSERT_TRUE(t == Test(idx, 'b'));
     }
 
     // Pop the last element from array (checks if destructor is called)
     array.pop_back();
 
-    MBED_HOSTTEST_ASSERT(array.get_num_elements() == initial_capacity - 1);
-    MBED_HOSTTEST_ASSERT(array.get_num_zones() == 1);
+    TEST_ASSERT_TRUE(array.get_num_elements() == initial_capacity - 1);
+    TEST_ASSERT_TRUE(array.get_num_zones() == 1);
+    }
+    TEST_ASSERT_TRUE(Test::inst_count == 0);
 }
 
-void app_start(int, char**) {
-    MBED_HOSTTEST_TIMEOUT(5);
-    MBED_HOSTTEST_SELECT(default);
-    MBED_HOSTTEST_DESCRIPTION(mbed-util array  test);
-    MBED_HOSTTEST_START("MBED_UTIL_ARRAY_TEST");
+static status_t test_setup(const size_t number_of_cases) {
+    GREENTEA_SETUP(5, "default_auto");
 
-    test_pod(); // test with "plain old data"
-    test_non_pod(); // test with complex data
-    MBED_HOSTTEST_ASSERT(Test::inst_count == 0);
+    return greentea_test_setup_handler(number_of_cases);
+}
+
+status_t greentea_failure_handler(const Case *const source, const failure_t reason) {
+    greentea_case_failure_abort_handler(source, reason);
+    return STATUS_CONTINUE;
```

Comment:
Sorry, that's a load of BS.
The `greentea_case_failure_abort_handler` won't die itself, the *harness* might decide to die, [depending on where the failure occured](https://github.com/PrzemekWirkus/utest/blob/devel_transport/source/harness.cpp#L133)
However, since you return `STATUS_CONTINUE` anyways, this is perfectly safe to call, however, really confusing ;-P